src/Symfony/Component/Serializer/Mapping/Factory/ClassResolverTrait.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Mapping\Factory;
+
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
+
+/**
+ * Resolves a class name.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+trait ClassResolverTrait
```

Comment:
In that case they could use the ``ClassUtils`` of doctrine/common. 
IMO the behaviour of this trait is easily reproductible and should not be accessible publicly (at least in a trait which is not related directly to class management).